lang/cs/Org.Apache.REEF.IO/PartitionedData/Random/RandomInputPartition.cs

```diff
@@ -52,13 +53,18 @@ public string Id
             get { return _id; }
         }
 
-        public void Cache()
+        public CacheLevel Cache(CacheLevel level)
         {
             lock (_lock)
             {
+                if (level > CacheLevel.InMemoryMaterialized)
```

Comment:
Shouldn't we throw an error here saying that we support only CacheLevel.InMemoryMaterialized